ml_perf/reference_implementation.py

```diff
@@ -18,59 +18,53 @@
 sys.path.insert(0, '.')  # nopep8
 
 import logging
-import numpy
+import numpy as np
 import os
 import random
 import re
 import shutil
 import subprocess
-import tensorflow
+import tensorflow as tf
 import utils
 
 from absl import app, flags
-from rl_loop import example_buffer, fsdb, shipname
+from rl_loop import example_buffer, fsdb
 
 flags.DEFINE_string('engine', 'tf', 'Engine to use for inference.')
 
 FLAGS = flags.FLAGS
 
 
+# Note that we rely on the iteration number being the first part of the model
+# name so that the training chunks sort correctly.
 class State:
 
-  # TODO(tommadams): remove the random naming scheme and just use the generation
-  # number.
-  _NAMES = ['bootstrap'] + random.Random(0).sample(shipname.NAMES,
-                                                   len(shipname.NAMES))
-
   def __init__(self):
     self.iter_num = 0
-    self.play_model_num = 0
-    self.play_model_name = self.play_output_name
-    self.train_model_num = 1
+    self.gen_num = 0
 
-  @property
-  def play_output_name(self):
-    return '%06d-%s' % (self.iter_num, self._NAMES[self.play_model_num])
+    # We start playing using a random model.
+    # After the first round of selfplay has been completed, the engine is
+    # updated to FLAGS.engine.
+    self.engine = 'random'
 
-  @property
-  def play_model_path(self):
-    return os.path.join(fsdb.models_dir(), self.play_model_name)
+    self.best_model_name = 'random'
 
   @property
-  def train_model_name(self):
-    return '%06d-%s' % (self.iter_num, self._NAMES[self.train_model_num])
+  def output_model_name(self):
```

Comment:
around here might be a good place for a comment about the naming scheme.

Reply:
added it to the top of the State class